scale/data/models.py

```diff
@@ -365,26 +457,23 @@ class Meta(object):
 class DataSetFileManager(models.Manager):
     """Manages the datasetfile model"""
 
-    def create_dataset_files(self, dataset, data):
+    def create_dataset_files(self, dataset, data, existing_scale_ids):
         """Creates dataset files for the given dataset and data"""
 
-        file_ids = []
         datasetfiles = []
         for i in data.values.keys():
             v = data.values[i]
             if type(v) is FileValue:
-                file_ids.extend(v.file_ids)
                 for id in v.file_ids:
-                    exists = DataSetFile.objects.filter(dataset=dataset, scale_file_id=id).count()
-                    if exists:
+                    if id in existing_scale_ids:
```

Comment:
I'm guessing this is to make sure we don't get duplicates?

Reply:
Yes, and looking at it I realized I forgot to append to that list as I go along so there aren't any duplicates in the files created from the data list.